src/planner/planner.c

```diff
@@ -678,35 +675,45 @@ get_or_add_baserel_from_cache(Oid chunk_reloid, Oid parent_reloid)
 		return entry;
 	}
 
-	/*
-	 * This reloid is not in the chunk cache, so do the full metadata
-	 * lookup.
-	 */
-	int32 hypertable_id = 0;
-	int32 chunk_status = 0;
-	if (ts_chunk_get_hypertable_id_and_status_by_relid(chunk_reloid, &hypertable_id, &chunk_status))
+	if (OidIsValid(parent_reloid))
 	{
-		/*
-		 * This is a chunk. Look up the hypertable for it.
-		 */
-		if (OidIsValid(parent_reloid))
+		ht = ts_planner_get_hypertable(parent_reloid, CACHE_FLAG_CHECK);
+
+#ifdef USE_ASSERT_CHECKING
+		/* Sanity check on the caller-specified hypertable reloid. */
+		int32 parent_hypertable_id = ts_chunk_get_hypertable_id_by_relid(chunk_reloid);
+		if (parent_hypertable_id != INVALID_HYPERTABLE_ID)
 		{
-			/* Sanity check on the caller-specified hypertable reloid. */
-			Assert(ts_hypertable_id_to_relid(hypertable_id) == parent_reloid);
+			Assert(ts_hypertable_id_to_relid(parent_hypertable_id) == parent_reloid);
+
+			if (ht != NULL)
+			{
+				Assert(ht->fd.id == parent_hypertable_id);
+			}
 		}
-		else
+#endif
+	}
+	else
+	{
+		/* Hypertable reloid not specified by the caller, look it up by
+		 * an expensive metadata scan.
+		 */
+		int32 hypertable_id = ts_chunk_get_hypertable_id_by_relid(chunk_reloid);
+
+		if (hypertable_id != INVALID_HYPERTABLE_ID)
 		{
 			/* Hypertable reloid not specified by the caller, look it up. */
 			parent_reloid = ts_hypertable_id_to_relid(hypertable_id);
+			Assert(OidIsValid(parent_reloid));
+
+			ht = ts_planner_get_hypertable(parent_reloid, CACHE_FLAG_NONE);
+			Assert(ht != NULL);
+			Assert(ht->fd.id == hypertable_id);
 		}
-		ht = ts_planner_get_hypertable(parent_reloid, CACHE_FLAG_NONE);
-		Assert(ht != NULL);
-		Assert(ht->fd.id == hypertable_id);
 	}
 
 	/* Cache the result. */
 	entry->ht = ht;
-	entry->chunk_status = chunk_status;
```

Comment:
... which means that `ht` could be NULL here, which can lead to a crash if an attempt to dereference `ht` is made, which is done in `classify_relation`, immediately after this return.

Reply:
... ah, mea culpa, that does not seem possible, but please check that there is no risk of a crash on a release build.

Reply:
@mkindahl We call `get_or_add_baserel_from_cache` two times in our code. It is called in `classify_relation` [[1](https://github.com/timescale/timescaledb/blob/462aa4faeaf443a9f68d876ae9fd9e0d796a6f91/src/planner/planner.c#L757), [2](https://github.com/timescale/timescaledb/blob/462aa4faeaf443a9f68d876ae9fd9e0d796a6f91/src/planner/planner.c#L795)]. Both times,`ht` is checked for null value. Do you have anything specific in mind I should do here?

Reply:
No, I saw the dereference first and assumed that was the pointer from the cache, but it was not. Nothing to do here.